filesystems/memory.py

```diff
@@ -267,7 +270,10 @@ class _Link(object):
     def __getitem__(self, name):
         return self._entry_at()[name]
 
-    def create_directory(self, path, with_parents):
+    def create_directory(self, path, with_parents, allow_existing):
+        # TODO: should this stat stuff really be used here?
```

Comment:
I think you just want:

```python
if allow_existing:
    return self._entry_at(path=path).create_directory(...)
```

no?

Reply:
I keep forgetting to look at attributes for 'methods'...  I mean really, I do this in my own projects so it's not like it's a foreign concept.  :[  Though, a type hint of callable could be a nice touch.